runtime/metricsview/executor.go

```diff
@@ -320,3 +322,150 @@ func (e *Executor) Export(ctx context.Context, qry *Query, executionTime *time.T
 		"args": args,
 	})
 }
+
+// SearchQuery executes the provided query against the metrics view.
+func (e *Executor) Search(ctx context.Context, qry *SearchQuery, executionTime *time.Time) ([]SearchResult, error) {
+	if !e.security.CanAccess() {
+		return nil, runtime.ErrForbidden
+	}
+
+	// Generate a metricsview.Query and build a AST
+	// This is a hacky implementation since both metricsview.Query and AST are designed for aggregate queries.
+	// TODO :: Refactor the code and extract common functionality from metricsview.Query and AST and write SearchQuery to underlying SQL/Native druid query directly.
+
+	if e.olap.Dialect() == drivers.DialectDruid {
+		dimensions := make([]Dimension, len(qry.Dimensions))
+		for i, d := range qry.Dimensions {
+			dimensions[i] = Dimension{Name: d}
+		}
+		q := &Query{
+			MetricsView: qry.MetricsView,
+			Dimensions:  dimensions,
+			Where:       qry.Where,
+			Having:      qry.Having,
+			TimeRange:   qry.TimeRange,
+			Limit:       qry.Limit,
+		}
```

Comment:
nit: Can you add all fields in the struct, even the `nil` ones? Since it would be quite bad if we add a new field and don't add it everywhere, doing this will reduce the chance of errors. Btw, I'm looking into setting up this package to force setting struct members for specific structs: https://github.com/GaijinEntertainment/go-exhaustruct

Reply:
Seems somewhat painful to specify nil fields if there are more than few. 
But could be an easy solution to make sure fields are not missed.

Reply:
Not thinking about adding it for all structs, that would be too painful, only for critical ones (like project update) – it has `//exhaustruct:enforce` to selectively add to specific structs